# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/base/nsCaret.cpp
# Commit: b32811ea1c61
# Full Hash: b32811ea1c61e1891454d826f7ba003416c9eb57
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2024-03-11 21:13:39
# Regressor Bug: 1860328
# File Overlap Count: 2
# Description:
#   Bug 1860328 - Fix a crash which a caret frame misses a MarkNeedsDisplayItemRebuild() call r=emilio
#   
#   Generally, we mark a caret frame for display first, and then
#   nsCaret tracks this frame in nsCaret::SchedulePaint to call
#   MarkNeedsDisplayItemRebuild() accordingly. However, it's possible
# ==============================================================================

diff -r caab023e5d2d -r b32811ea1c61 layout/base/nsCaret.cpp
--- a/layout/base/nsCaret.cpp	Mon Mar 11 13:41:27 2024 +0000
+++ b/layout/base/nsCaret.cpp	Mon Mar 11 13:41:28 2024 +0000
@@ -413,7 +413,7 @@
   nsIFrame* frame = GetFrameAndOffset(selection, mOverrideContent,
                                       mOverrideOffset, &frameOffset);
   if (!frame) {
-    mLastCaretFrame = nullptr;
+    SetLastCaretFrame(nullptr);
     return;
   }
 
@@ -421,7 +421,7 @@
     frame = cb;
   }
 
-  mLastCaretFrame = frame;
+  SetLastCaretFrame(frame);
   frame->SchedulePaint();
 }
 